bertopic/representation/_langchain.py

```diff
@@ -186,27 +227,49 @@ def extract_topics(
-
-            inputs = [{"input_documents": docs, "question": prompt} for docs, prompt in zip(chain_docs, prompts)]
-
-        else:
-            inputs = [{"input_documents": docs, "question": self.prompt} for docs in chain_docs]
+        # Extract keywords from the topics and format them as a string
+        formatted_keywords_list = []
+        for topic in topics:
+            keywords = list(zip(*topics[topic]))[0]
+            formatted_keywords_list.append(", ".join(keywords))
+
+        # self.chain must accept DOCUMENTS as a mandatory input key and KEYWORDS as an optional input key
+        # We always pass both keys to the chain, and the chain can choose to use them or not
+        # Documents are passed as a list of LangChain Document objects, it is up to the chain to format them into a string
+        inputs = [
+            {"DOCUMENTS": docs, "KEYWORDS": formatted_keywords}
+            for docs, formatted_keywords in zip(chain_docs, formatted_keywords_list)
+        ]
 
-        # `self.chain` must return a dict with an `output_text` key
-        # same output key as the `StuffDocumentsChain` returned by `load_qa_chain`
+        # self.chain must return a string label or a list of string labels for each input
         outputs = self.chain.batch(inputs=inputs, config=self.chain_config)
-        labels = [output["output_text"].strip() for output in outputs]
 
-        updated_topics = {
-            topic: [(label, 1)] + [("", 0) for _ in range(9)] for topic, label in zip(repr_docs_mappings.keys(), labels)
-        }
+        # Process outputs from the chain - can be either strings or lists of strings
+        updated_topics = {}
+        for topic, output in zip(repr_docs_mappings.keys(), outputs):
+            # Each output can be either:
+            # - A single string representing the main topic label
+            # - A list of strings representing multiple related labels
+            if isinstance(output, str):
+                # For string output: use it as the main label (weight=1)
+                # and pad with 9 empty strings (weight=0)
+                labels = [(output.strip(), 1)] + [("", 0) for _ in range(9)]
+            else:
+                # For list output:
+                # 1. Convert all elements to stripped strings
+                # 2. Take up to 10 elements
+                # 3. Assign decreasing weights from 1.0 to 0.1
+                # 4. Pad with empty strings if needed to always have 10 elements
+                clean_outputs = [str(label).strip() for label in output]
+                top_labels = clean_outputs[:10]
+
+                # Create (label, weight) pairs with decreasing weights
+                labels = [(label, 1.0 - (i * 0.1)) for i, label in enumerate(top_labels)]
+
+                # Pad with empty strings if we have less than 10 labels
+                if len(labels) < 10:
+                    labels.extend([("", 0.0) for _ in range(10 - len(labels))])
```

Comment:
This is interesting. The output of any generative model in BERTopic is meant to be the same as you did above:

```python
[("my label", 1), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0)]
```

but what wasn't implemented before is that you could also generate a list of keywords/labels. Do you have an example of when this piece of code would be executed? When is the output a list rather than a single string? 

Also, I'm a bit hesistant giving decreasing weights rather than all 1s since (if I'm not mistaken) the weights do not have any meaning.

Reply:
To be honest, I wasn't entirely sure about the meaning behind the list of tuples and the weights so I just kept the old behaviour (which you have provided an example of) and, given that format, I assumed it could be extended to allow for generated lists of labels. In the case of lists, I don't mind setting the weights to all 1s (again, I didn't research the meaning of that format for representations).

The need to allow for lists stemmed from a use-case I had where I used a custom chain to generate topic labels in several languages with the current implementation. Since the current implementation does not allow for lists, I concatenated all elements of the list generated by the chain into a single string with a specific separator so that it could be split later. Allowing for lists in the chain output would make it possible to avoid this. Granted, this may be overkill 😄 

I provided an example of basic usage, basic usage with custom prompt and advanced usage with different types of list outputs in this thread. Maybe looking at that code and its output would make it more explicit 😃

Reply:
> To be honest, I wasn't entirely sure about the meaning behind the list of tuples and the weights so I just kept the old behaviour (which you have provided an example of) and, given that format, I assumed it could be extended to allow for generated lists of labels. In the case of lists, I don't mind setting the weights to all 1s (again, I didn't research the meaning of that format for representations).

There isn't actually any LLM implemented in BERTopic currently that returns a list of labels. They all return a single label. Although I like the idea of returning multiple labels, I wou suggest removing this here considering this might be a bit out of scope for this PR. 

> The need to allow for lists stemmed from a use-case I had where I used a custom chain to generate topic labels in several languages with the current implementation. Since the current implementation does not allow for lists, I concatenated all elements of the list generated by the chain into a single string with a specific separator so that it could be split later. Allowing for lists in the chain output would make it possible to avoid this. Granted, this may be overkill 😄

Hmmm, this is rather interesting use case that I haven't seen before. Now you make hesistate on the best course of action here... Nevermind what I said above, let's keep this and make sure they all have values of 1 instead of a decreasing value. Since the weights are currently meaningless (and quite frankly not used) in the LLM-setting, we can just set them to 1.

> I provided an example of basic usage, basic usage with custom prompt and advanced usage with different types of list outputs in this thread. Maybe looking at that code and its output would make it more explicit 😃

Wow, these are amazing examples! Thanks for sharing.

Reply:
I think that the nice thing about the list behaviour is that it fits nicely with what is already implemented as returning
`[("my label", 1), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0)]`
or
`[("my first label", 1), ("my second label", 1), ("my third label", 1), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0), ("", 0)]`
seems consistent. In addition, with the basic `llm` + `prompt` usage only the first behaviour happens. This just gives more flexibility for using custom chains.

I have adapted the code so that the weight is 1 for all labels.